src/common/regions.ts

```diff
@@ -116,27 +121,65 @@ const counties: County[] = chain(state_county_map_dataset)
 class RegionDB {
   constructor(private regions: Region[]) {}
 
-  private filterByType(regionType: RegionType) {
-    return this.all().filter(region => region.regionType === regionType);
-  }
-
   findByFipsCode(fipsCode: FipsCode): Region | null {
     const region = this.regions.find(region => region.fipsCode === fipsCode);
     return region || null;
   }
 
+  findStateByUrlParams(stateId: string): State | null {
+    const foundState = this.states().find(
+      state =>
+        equalLower(state.urlSegment, stateId) ||
+        equalLower(state.stateCode, stateId),
```

Comment:
Perhaps add a comment that this supports "legacy" URLs that just included the state code in the URL?